### PR TITLE
Release v1.23.0

### PR DESCRIPTION
**Pull request by jmrodri**
**Description of the change:**
Release v1.23.0 of `operator-sdk`

**Motivation for the change:**

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by everettraven — APPROVED** [review]
/lgtm

**Review by theishshah — APPROVED** [review]
/lgtm pending CI

**Comment by theishshah** [comment]
new /lgtm

File: changelog/generated/v1.23.0.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- **Breaking change**: (ansible/v1,helm/v1) Add support for Apple Silicon(`darwin/arm64`) and upgrade kustomize version from `v3.8.7` to `v4.5.5`. [Release Notes](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.0.0). ([#5965](https://github.com/operator-framework/operator-sdk/pull/5965))
+- Update the supported OLM versions to 0.20.0, 0.21.2, 0.22.0. ([#6000](https://github.com/operator-framework/operator-sdk/pull/6000))
+
+### Deprecations
+
+- Legacy layout "Kubebuilder 2.x" provided via the go/v2 plugin is officially deprecated. Be aware that since `Apr 2021`, the default layout produced is done via the `go/v3`. ([More info](https://github.com/kubernetes-sigs/kubebuilder/discussions/2842)) Check your `PROJECT` file to make sure that your project layout is `go.kubebuilder.io/v3`, if not please follow the [migration guide](https://sdk.operatorframework.io/docs/building-operators/golang/migration/) to upgrade your project. ([#5965](https://github.com/operator-framework/operator-sdk/pull/5965))
```
Comment (asmacdo) [inline changelog/generated/v1.23.0.md:25]: + dropped olm support 0.19.1

File: changelog/generated/v1.23.0.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### Bug Fixes
+
+- For Hybrid Helm (hybrid.helm/v1-alpha) language based operators: Bump Dockerfile Go version to 1.18. ([#5772](https://github.com/operator-framework/operator-sdk/pull/5772))
+- Fix Makefile targets to not download binaries when they are found already. ([#5965](https://github.com/operator-framework/operator-sdk/pull/5965))
+- Remove the option to call `kustomize/v1` plugin alone. If you are looking to use this project as a base for you to generate other plugins, please use [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) instead. ([#5965](https://github.com/operator-framework/operator-sdk/pull/5965))
+- Fix operator-sdk run bundle and upgrade bundle subCommands to allow they to work against Kubernetes versions < 1.19 and vendors like Openshift. ([#5973](https://github.com/operator-framework/operator-sdk/pull/5973))
```
Comment (asmacdo) [inline changelog/generated/v1.23.0.md:32]:
```suggestion
- Fix operator-sdk run bundle and upgrade bundle subCommands to allow them to work against Kubernetes versions < 1.19 and vendors like Openshift. ([#5973](https://github.com/operator-framework/operator-sdk/pull/5973))
```

**Review by asmacdo — APPROVED** [review]
some changelog nits, no need for re-review from me 
/lgtm

**Review by laxmikantbpandhare — APPROVED** [review]
/lgtm

one of the links is not accessible that's why failing with lint errors and I guess that will be fine.

**Pull request merged**